src/components/Footer/LoginButton.js

```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+import { Link } from 'react-router-dom'
+import FontAwesome from 'react-fontawesome'
+import styles from './styles'
+
+const routeIfAuth = firebase =>
+  firebase.auth().currentUser
+    ? '/admin'
+    : '/login'
+
+const LoginButton = ({firebase}) =>
```

Comment:
does `LoginButton` need the entire `firebase` object? with simple components like this, it can be nice to work with a prop like `isAuthenticated` and move the computation of that value further up to a container or `connect`.

Reply:
How would that look? Would the footer object get connected with firebase then and just pass down to child?

Reply:
I might approach it by writing a HOC whose job is just to take a component and return a component with that prop connected. I think you could do something like
```js
const connectIsAuthenticated = compose( // from recompose
  firebaseConnect([
    'auth'
  ]),
  withProps( // from recompose
    props => ({ isAuthenticated: Boolean(props.firebase.auth().currentUser) })
  )
)